### PR TITLE
[WIP][19.09] New outputs/ directory needs to be mounted rw.

### PR DESCRIPTION
**Pull request by jmchilton**
Added in ad573b215802aaaa9d09e1d2e3e2974e79295dfe.

### CODE REVIEW

**Review by mvdbeek — DISMISSED** [review]

File: lib/galaxy/tool_util/deps/container_classes.py [hunk 1]
```diff
@@ -178,7 +178,7 @@ def add_var(name, value):
             if self.job_info.tool_directory:
                 defaults += ",$tool_directory:default_ro"
             if self.job_info.job_directory:
-                defaults += ",$job_directory:default_ro"
+                defaults += ",$job_directory:default_ro,$job_directory/outputs:rw"
```
Comment (mvdbeek) [inline lib/galaxy/tool_util/deps/container_classes.py:181]: So do singularity and docker handle that correctly ? I think there's a change they may not :/

Comment (jmchilton) [reply lib/galaxy/tool_util/deps/container_classes.py:181]: The working directory uses the same trick though right? Everything is broken if this doesn't work 😿.

Comment (mvdbeek) [reply lib/galaxy/tool_util/deps/container_classes.py:181]: I have to admit that I cheated when I deployed singularity and just mounted everything rw, but I can check

Comment (jmchilton) [reply lib/galaxy/tool_util/deps/container_classes.py:181]: I think Singularity is a known quantity and we work around it?

> Galaxy allows the mount try "default_ro" which translates to ro for Docker and ro for Singularity iff no subdirectories are rw (Singularity does not allow ro parent directories with rw subdirectories).

Comment (mvdbeek) [reply lib/galaxy/tool_util/deps/container_classes.py:181]: ok, I think that's fine then.

**Comment by jmchilton** [comment]
We didn't add this feature in 19.09 :( , thanks for hunting that down @mvdbeek.

**Pull request closed without merging**